### PR TITLE
add RELEASING.md document

### PR DESCRIPTION
**Pull request by jpdahlke**

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+* [Versioning](#versioning)
+* [Release Types](#release-types)
+* [Publishing a Release](#publishing-a-release)
+
+## Introduction
+
+Release process uses GitHub actions and involves:
+- Creating a release branch to perform the following actions:
+  - Dry run a maven release
+  - Remove SNAPSHOT from the version and create scm tag
```
Comment (drivenflywheel) [inline RELEASING.md:16]:
```suggestion
  - Remove the "-SNAPSHOT" suffix from the version and create scm tag
```

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- From: `Branch: master` 
+- Type: `Release`
+- Suffix: Leave blank
+```
+Creates a branch called `release/<version>` and performs release. One commit is added to `master` only to increment to the next snapshot version.
+
+
+### Patch Release
+
+Releases may have bugs or vulnerabilities with a high enough severity that requires a patch release to fix.
```
Comment (drivenflywheel) [inline RELEASING.md:46]:
```suggestion
Releases may have bugs or vulnerabilities with a high enough severity that requires a patch release to fix. Patch releases can only be performed from a patch branch (described below).
```

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+- From: `Branch: patch/<version>`
+- Type: `Patch`
+- Suffix: Leave blank
+```
+Releases patch fixes from a branch called `patch/<version>`, i.e. `patch/8.0.x`. A release branch called `release/<version>` is created
+and the release is performed. When finished, the patch branch is deleted. No commits are made to `master`.
+
+### Iterative Release (Milestone)
+
+Iterative releases are not guaranteed to be stable, like snapshots, but allow for markers in the development stage and allow for testing and
```
Comment (drivenflywheel) [inline RELEASING.md:78]: Maybe?
```suggestion
Like snapshots, Iterative releases are not guaranteed to be stable, but allow for markers in the development stage and allow for testing and
```

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
Minor suggestions.  They can be dismissed with little impact.

**Review by cfkoehler — APPROVED** [review]

**Comment by ldhardy** [comment]
I know this was already merged, but these instructions didn't include any details about the preferred way the patch makes it into the master branch.   Or was the idea that the change is made in master and then cherry picked into the patch branch?

**Comment by cfkoehler** [comment]
> I know this was already merged, but these instructions didn't include any details about the preferred way the patch makes it into the master branch. Or was the idea that the change is made in master and then cherry picked into the patch branch?

That is the preferred way. But no reason we can’t be explicit about it in the doc. I can add a section describing that later today and put up a PR

**Pull request merged**